### PR TITLE
Update dependency go modules for k8s v1.28.0-rc.0

### PR DESCRIPTION
**Pull request by Sneha-at**
Ran kubernetes-csi/release-tools go-get-kubernetes.sh -p 1.28.0-rc.0

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -42,27 +42,37 @@ require (
...
-	k8s.io/kube-openapi v0.0.0-20230501164219-8b0f38b5fd1f // indirect
-	k8s.io/utils v0.0.0-20230209194617-a36077c30491 // indirect
+	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
+	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace k8s.io/api => k8s.io/api v0.28.0-rc.0
```
Comment (sunnylovestiramisu) [inline go.mod:70]: Remove these replace statements?

File: go.mod [hunk 2]
```diff
@@ -11,9 +11,9 @@ require (
 	github.com/kubernetes-csi/csi-lib-utils v0.14.0
 	github.com/kubernetes-csi/csi-test/v4 v4.4.0
 	google.golang.org/grpc v1.57.0
-	k8s.io/api v0.27.4
-	k8s.io/apimachinery v0.27.4
-	k8s.io/client-go v0.27.4
+	k8s.io/api v0.28.0-rc.0
+	k8s.io/apimachinery v0.28.0-rc.0
+	k8s.io/client-go v0.28.0-rc.0
 	k8s.io/csi-translation-lib v0.27.4
```
Comment (sunnylovestiramisu) [inline go.mod:17]: This one also has https://github.com/kubernetes/csi-translation-lib/tree/v0.28.0-rc.0 it should not be 0.27.4, something is wrong with the script.

Comment (Sneha-at) [reply go.mod]: I tried running `go mod tidy` but even that added version `v0.27.4`. For [livenessprobe](https://github.com/kubernetes-csi/livenessprobe/pull/201#discussion_r1283719137) manually updating this to v0.28-rc.0 caused failure so did not try this here. I have pushed a commit now to update `csi-translation-lib` to  v0.28-rc.0, hoping it doesn't error here.

**Pull request merged**